pghoard/basebackup/delta.py

```diff
@@ -75,7 +78,23 @@ def __init__(
         self.data_file_format = data_file_format
 
     def _snapshot(self, snapshotter: Snapshotter) -> SnapshotResult:
-        snapshotter.snapshot(reuse_old_snapshotfiles=False)
+        def progress_callback(msg: ProgressStep, progress_data: ProgressMetrics):
+            key = "snapshot_progress"
+            persisted_progress = PersistedProgress.read(self.metrics)
+            progress_info = persisted_progress.get(key)
+            if progress_data["handled"] > progress_info.current_progress:
+                progress_info.update(progress_data["handled"])
+                persisted_progress.write(self.metrics)
+                self.metrics.gauge("pghoard.seconds_since_backup_progress_stalled", 0)
```

Comment:
You need to put the phase in the tags, so we can differentiate between the various snapshot ones and the upload one.

Reply:
will add the phase tags